multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/util/NamespaceValidationUtil.java

```diff
@@ -1,35 +1,46 @@
 package org.cloudfoundry.multiapps.controller.core.util;
 
+import static java.text.MessageFormat.format;
+
+import java.util.Map;
+import java.util.Objects;
+
 import org.cloudfoundry.multiapps.common.util.MapUtil;
 import org.cloudfoundry.multiapps.controller.core.Messages;
 import org.cloudfoundry.multiapps.controller.core.model.SupportedParameters;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.Map;
-import java.util.Objects;
-
-import static java.text.MessageFormat.format;
-
 public class NamespaceValidationUtil {
     private static final Logger LOGGER = LoggerFactory.getLogger(NamespaceValidationUtil.class);
     private String namespace;
     private boolean applyNamespaceGlobalLevel;
     private Boolean applyNamespaceProcessVariable;
+    private boolean applyNamespaceAsSuffixGlobalLevel;
+    private Boolean applyNamespaceAsSuffixProcessVariable;
 
     public NamespaceValidationUtil() {
     }
 
-    public NamespaceValidationUtil(String namespace, boolean applyNamespaceGlobalLevel, Boolean applyNamespaceProcessVariable) {
+    public NamespaceValidationUtil(String namespace, boolean applyNamespaceGlobalLevel, Boolean applyNamespaceProcessVariable,
+                                   boolean applyNamespaceAsSuffixGlobalLevel, Boolean applyNamespaceAsSuffixProcessVariable) {
         this.namespace = namespace;
         this.applyNamespaceGlobalLevel = applyNamespaceGlobalLevel;
         this.applyNamespaceProcessVariable = applyNamespaceProcessVariable;
+        this.applyNamespaceAsSuffixGlobalLevel = applyNamespaceAsSuffixGlobalLevel;
+        this.applyNamespaceAsSuffixProcessVariable = applyNamespaceAsSuffixProcessVariable;
     }
 
     public String getNamespace() {
         return namespace;
     }
 
+    protected boolean shouldApplyNamespaceAsSuffix(final Map<String, Object> relatedParameters) {
+        Boolean applyNamespaceAsSuffixModuleLevel = MapUtil.parseBooleanFlag(relatedParameters, SupportedParameters.APPLY_NAMESPACE_AS_SUFFIX, null);
+        return Objects.requireNonNullElse(applyNamespaceAsSuffixProcessVariable,
```

Comment:
it would be mroe readable and faster if you just add 
if (applyNamespaceAsSuffixProcessVariable != null) return applyNamespaceAsSuffixProcessVariable
at the start